aws-android-sdk-cognitoidentityprovider/src/main/java/com/amazonaws/mobileconnectors/cognitoidentityprovider/CognitoUser.java

```diff
@@ -3163,8 +3163,9 @@ public void run() {
             nextTask = deviceSrpAuthentication(clientMetadata, challenge, callback, runInBackground);
         } else if (CognitoServiceConstants.CHLG_TYPE_NEW_PASSWORD_REQUIRED.equals(challengeName)) {
             final NewPasswordContinuation newPasswordContinuation = new NewPasswordContinuation(
-                    cognitoUser, context, usernameInternal, clientId, secretHash, challenge,
-                    runInBackground, callback);
+                    cognitoUser, context, usernameInternal, clientId,
+                    CognitoSecretHash.getSecretHash(usernameInternal, clientId, clientSecret),
```

Comment:
Does the `secretHash` itself need this update? Why only apply it here?

Reply:
Great question! Maybe but I'm not sure. To answer this, we need more information
from the server-side about the usage of the client secret-hash.

Is there any use for the client secret-hash after the AuthenticationContinuation? or we can override it?

I mean by client secret-hash is this.
client = `email`, `clientId`, `clientSecret`.

And I'm applying it only here because I am working on the reset password challenge
and faced this problem. But it could be relevant for other challenges as well worth a check.

Reply:
Great question! Maybe but I'm not sure. To answer this, we need more information
from the server-side about the usage of the client secret-hash.

Is there any use for the client secret-hash after the AuthenticationContinuation? or we can override it?

I mean by client secret-hash is this.
client = `email`, `clientId`, `clientSecret`.

And I'm applying it only here because I am working on the new password required challenge
and faced this problem. But it could be relevant for other challenges as well worth a check.

Reply:
I'll forward this to the Cognito team and see if anyone has guidance for us.

Reply:
Cognito team reviewed this internally and approved.